src/storage/mvcc/reader/point_getter.rs

```diff
@@ -223,94 +223,102 @@ impl<S: Snapshot> PointGetter<S> {
-                    self.statistics.write.processed_keys += 1;
-
-                    if self.omit_value {
-                        return Ok(Some(vec![]));
+            loop {
+                // We may seek to another key. In this case, it means we cannot find the specified key.
+                {
+                    let cursor_key = self.write_cursor.key(&mut self.statistics.write);
+                    if !Key::is_user_key_eq(cursor_key, user_key.as_encoded().as_slice()) {
+                        return Ok(None);
                     }
-                    match write.short_value {
-                        Some(value) => {
-                            // Value is carried in `write`.
-                            return Ok(Some(value.to_vec()));
+                }
+
+                let write = WriteRef::parse(self.write_cursor.value(&mut self.statistics.write))?;
+
+                if !write.check_gc_fence_as_latest_version(self.ts) {
+                    return Ok(None);
+                }
+
+                match write.write_type {
+                    WriteType::Put => {
+                        self.statistics.write.processed_keys += 1;
+
+                        if self.omit_value {
+                            return Ok(Some(vec![]));
                         }
-                        None => {
-                            let start_ts = write.start_ts;
-                            return Ok(Some(self.load_data_from_default_cf(start_ts, user_key)?));
+                        match write.short_value {
+                            Some(value) => {
+                                // Value is carried in `write`.
+                                return Ok(Some(value.to_vec()));
+                            }
+                            None => {
+                                let start_ts = write.start_ts;
+                                return Ok(Some(
+                                    self.load_data_from_default_cf(start_ts, user_key)?,
+                                ));
+                            }
                         }
                     }
+                    WriteType::Delete => {
+                        return Ok(None);
+                    }
+                    WriteType::Lock | WriteType::Rollback => {
+                        // Continue iterate next `write`.
+                    }
                 }
-                WriteType::Delete => {
+
+                if !self.write_cursor.next(&mut self.statistics.write) {
                     return Ok(None);
                 }
-                WriteType::Lock | WriteType::Rollback => {
-                    // Continue iterate next `write`.
-                }
             }
-
-            if !self.write_cursor.next(&mut self.statistics.write) {
-                return Ok(None);
-            }
-        }
+        };
+        let timer = Instant::now();
+        let result = f();
+        self.statistics.wall_time += timer.elapsed();
```

Comment:
I think this will have a dramatic impact on the performance.

Reply:
How about using a coarse time at the outside? For example, in batch point get, this function will be called multiple times in a loop, and actually we only need to measure at the most outside.

Reply:
How about using a coarse time at the outside? For example, in batch point get, this function will be called multiple times in a loop, and actually we only need to measure at the most outside.